apps/public-reference/cypress/integration/pages/application/contact/name.ts

```diff
@@ -0,0 +1,50 @@
+describe("Form contact/name", function () {
+  it("Renders the /contact/name form", function () {
+    cy.visit("/applications/contact/name")
+    cy.get("form").should("be.visible")
+    cy.location("pathname").should("include", "/applications/contact/name")
+  })
+
+  it("Should display initial form errors", function () {
+    // try to trigger form
+    cy.get("button").contains("Next").click()
+
+    // check errors
+    cy.get('span[id="applicant.firstName-error"]').contains("Please enter a First Name")
+    cy.get('span[id="applicant.lastName-error"]').contains("Please enter a Last Name")
+    cy.get(".error-message").contains("Please enter a valid Date of Birth")
+    cy.get('span[id="applicant.emailAddress-error"]').contains("Please enter an email address")
```

Comment:
This seems pretty tightly coupled to the implementation to me (grabbing a `span` by the `id`) - if we change what the errors look like (either the format of the ID or the HTML element), these tests will fail. Maybe a better way would just be to have a selector for the root error block, and then search for the test within child elements? We could also make the text slightly less specific (e.g. just search for "First Name" instead of the full string in case we want to update the exact text later, but I don't have a strong feeling about that either way.

Reply:
Agree, I added HTML selectors, because I had a weird error... but I noticed it occurs because of ID's include dots.
I created 2 support commands to get element by ID & to click the next button - it should be better.

I think e.g. _applicant.lastName-error_ (generated by component automatically) is secure - I don't see reason, why we'll change it, but even - if we may want to change this behavior, we just can create prop for the error message ID in the future.

The best approach will be to create custom data- attribute, but it obliges us to update the current form so probably we don't want it.

I don't love the approach with search partial string, for me test should test the exact string or just test element visibility.